pkg/controllermanager/leaderelection/leaderelection.go

```diff
@@ -76,6 +77,7 @@ func NewFederationLeaderElector(
 		LeaseDuration: 15 * time.Second,
 		RenewDeadline: 10 * time.Second,
 		RetryPeriod:   5 * time.Second,
+		WatchDog:      healthzAdaptor,
```

Comment:
Do we need to `healthzAdaptor.SetLeaderElection(le)`?

Reply:
yes, fixed

Reply:
Actually, why don't we use `le.Check(maxTolerableExpiredDuration)` directly?

Reply:
that's what the adaptor is doing, it just wraps it in a way that is compatible with `healthz.Handler`